js/contribution.js

```diff
@@ -0,0 +1,135 @@
+const Contribution = require('../abs/contribution')
+const ledgerState = require('../browser-laptop/app/common/state/ledgerState')
+const ledgerStateData = require('../lib/data/ledger-state-funded.json')
+const ledgerStateOriginalData = require('../lib/data/ledger-state.json')
+const reconcileResponses = require('../lib/data/contribution-responses')
+const sinon = require('sinon')
+const now = new Date()
+class JS extends Contribution {
+  constructor () {
+    super()
+    this.paymentEnabled = true
+    this.paymentsAllowNonVerified = true
+  }
+
+  runBefore (mockery) {
+    this.before(mockery)
+    process.env.LEDGER_ENVIRONMENT = 'staging'
+    process.env.LEDGER_NO_FUZZING = true
+    process.env.LEDGER_NO_DELAY = true
+    sinon.useFakeTimers(now.getTime())
+    this.setStateFile()
+    this.ledger = require('../browser-laptop/app/browser/api/ledger')
+    this.ledgerStatuses = require('../browser-laptop/app/common/constants/ledgerStatuses')
+    this.loadStubs(['contribution'])
+  }
+
+  beforeEach () {
+    this.paymentEnabled = true
+    this.paymentsAllowNonVerified = true
+    this.ledger.setSynopsis(null)
+  }
+
+  after (mockery) {
+    mockery.deregisterAll()
+    mockery.disable()
+  }
+
+  afterEach (mockery) {
+    mockery.resetCache()
+    this.resetStateFile()
+    this.state = this.defaultAppState
+    this.ledger.resetModules()
+  }
+
+  clientSetup () {
+    this.client = this.ledger.getClient()
+    this.client.state.reconcileStamp = 1528948800000
+    this.client.credentials = {
+      persona: {
+        parameters: {
+          userId: '65ba48f6-591b-437c-ae74-219c8c5d4921'
+        }
+      }
+    }
+    const fCredential = reconcileResponses['reconcileCredentialFinalizeResponse']
+    const fPayload = reconcileResponses['reconcileCredentialSubmitResponse']
+    this.client.credentialFinalize = sinon.stub(this.client, 'credentialFinalize').callsFake(function (credential, verification, callback) {
+      callback(null, fCredential)
+    })
+    this.client.credentialSubmit = sinon.stub(this.client, 'credentialSubmit').callsFake(function (ballots, callback) {
+      callback(null, fPayload)
+    })
+    this.ledger.setClient(this.client)
+  }
+
+  setStateFile () {
+    this.stateFile = JSON.stringify(ledgerStateData)
+  }
+
+  resetStateFile () {
+    this.stateFile = JSON.stringify(ledgerStateOriginalData)
+  }
+
+  addPublishersToLedger () {
+    this.addPublisherToLedger('https://clifton.io', 1)
+    this.addPublisherToLedger('https://brianbondy.com', 2)
+    this.addPublisherToLedger('https://bumpsmack.com', 3)
+  }
+
+  addPublisherToLedger (location, tabId = true) {
+    this.setState(this.ledger.addNewLocation(this.state, location, tabId, false, true))
+    this.setState(this.ledger.pageDataChanged(this.state, {}, true))
+  }
+
+  staticizeBallots () {
+    let transaction = this.state.getIn(['ledger', 'info', 'transactions']).toJS()
+    transaction[0].ballots['brianbondy.com'] = 19
```

Comment:
Why is this done manually?

Reply:
Votes are randomized. So to keep snapshot tests from failing, we need to make votes constant across tests.